service/src/main/java/bio/terra/workspace/db/FolderDao.java

```diff
@@ -274,7 +291,7 @@ WITH RECURSIVE subfolders AS (
    * @return true if folder(s) are deleted.
    */
   @WriteTransaction
-  public boolean deleteFolderRecursive(UUID workspaceUuid, UUID folderId) {
+  public boolean deleteFoldersRecursive(UUID workspaceUuid, UUID folderId) {
```

Comment:
Can you add a comment like This cannot be used for root folder. For root folder, use deleteAllFolders().

Reply:
there is no root folder. It's a secret that FolderDao hold. also this folderId is nonNull.

Reply:
What do we store in DB for root folder, `0`? (Ie what is `parent_folder_id` for a top-level folder?) It's reasonable that someone could try to call this method for that, it doesn't hurt to add a comment and be clear.

Reply:
it's not stored. because 0 is not a uuid.

Reply:
Doesn't hurt to add a comment, in case someone's wondering how to recursively delete folders for root folder. :)

Reply:
to clarify, it's not stored as a row